### ISSUE TITLE
Incomplete list of sources/inputs in GrFN JSON output for PETPT.for

### ISSUE THREAD

**Issue by adarshp** [ml4ai/delphi#140]
The representation of the extracted `PETPT__assign__ALBEDO_1` function in the GrFN JSON output produced from `PETPT.for` is missing `MSALB` and `XHLAI` in its `sources` list.

```json
    {
      "name": "PETPT__assign__ALBEDO_1",
      "type": "assign",
      "target": "ALBEDO",
      "sources": [
        {
          "name": "math.exp",
          "type": "function"
        }
      ],
      "body": [
        {
          "type": "lambda",
          "name": "PETPT__lambda__ALBEDO_1",
          "reference": 13
        }
      ]
    },
```

Later on, in the `PETPT` container scope, the `input` list for `PETPT__assign__ALBEDO_1` contains only the `MSALB` variable, when it should ideally contain `XHLAI` as well.

```json
        {
          "name": "PETPT__assign__ALBEDO_1",
          "output": {
            "variable": "ALBEDO",
            "index": 3
          },
          "input": [
            {
              "variable": "MSALB",
              "index": 0
            }
          ]
        },
```

**Comment by pauldhein** [comment]
Here is another example of a function in the `pgm.json` file missing some inputs:
(`EEQ` and `TMAX` should be included as sources)
```json
{
      "name": "PETPT__assign__EO_2",
      "type": "assign",
      "target": "EO",
      "sources": [
        {
          "name": "math.exp",
          "type": "function"
        }
      ],
      "body": [
        {
          "type": "lambda",
          "name": "PETPT__lambda__EO_2",
          "reference": 21
        }
      ]
    },
```

**Comment by adarshp** [comment]
Closed by #146
